src/common/paddle/recruiter/processing.ts

```diff
@@ -47,14 +49,33 @@ export const createOpportunitySubscription = async ({
     return false;
   }
 
-  const opportunity: Pick<OpportunityJob, 'id'> = await con
-    .getRepository(OpportunityJob)
-    .findOneOrFail({
-      select: ['id'],
-      where: {
-        id: opportunity_id,
-      },
-    });
+  const opportunity: Pick<
+    OpportunityJob,
+    'id' | 'organizationId' | 'organization'
+  > = await con.getRepository(OpportunityJob).findOneOrFail({
+    select: ['id', 'organizationId', 'organization'],
+    where: {
+      id: opportunity_id,
+    },
+    relations: {
+      organization: true,
+    },
+  });
+
+  const organization = await opportunity.organization;
+
+  if (!organization) {
+    throw new Error(
+      'Opportunity does not have organization during payment processing, can not assign subscription, manual fixup needed',
+    );
+  }
+
+  if (
+    organization.recruiterSubscriptionFlags?.status ===
```

Comment:
We sure we have org at this point? as it's now a new flow where they can change lots of things basically?
I think so but might be some edge-cases?